### PR TITLE
refactor: rename merge-driver to mdsmith, generalize sections, add configurable files

### PR DESCRIPTION
**Pull request by jeduden**
## Summary

- Rename the git merge driver from `catalog` to `mdsmith` (`merge.mdsmith.*` in git config, `merge=mdsmith` in `.gitattributes`)
- Generalize conflict stripping to handle both `catalog` and `include` sections (any regenerable section type)
- Accept optional file arguments on `install`: `mdsmith merge-driver install [files...]` (default: PLAN.md, README.md)
- Use conflict state tracking so `=======` separators are only stripped between `<<<<<<<` and `>>>>>>>`, avoiding false positives with Markdown setext heading underlines
- Only strip `>>>>>>>` close markers when a matching `<<<<<<<` was seen inside the same section, preventing incorrect stripping when a conflict straddles a section boundary
- Preserve original file permissions via `os.Stat` instead of hardcoded 0644
- Check fix errors before writing to the ours temp file
- Update usage text and documentation across CLAUDE.md, AGENTS.md, `.github/copilot-instructions.md`, and README.md

## Key implementation details

`stripSectionConflicts` uses a two-level state machine: `inSection` tracks whether we are inside a regenerable section (`<!-- catalog`/`<!-- include` to `<!-- /catalog -->`/`<!-- /include -->`), and `inConflict` tracks whether we are inside a conflict block (`<<<<<<<` to `>>>>>>>`). Conflict markers are only stripped when both states are true, which correctly handles edge cases like conflicts straddling section boundaries and section markers appearing inside conflict blocks.

## Test plan

- [x] `go test ./...` — all tests pass
- [x] `go tool golangci-lint run` — no issues
- [x] `mdsmith check .` — no diagnostics
- [x] Existing merge-driver tests updated (`merge.catalog` -> `merge.mdsmith`)
- [x] `TestE2E_MergeDriver_Install_CustomFiles` — custom file args
- [x] `TestE2E_MergeDriver_IncludeConflict_Resolved` — include section conflicts
- [x] `TestE2E_MergeDriver_SetextInSection_Preserved` — setext headings survive
- [x] `TestE2E_MergeDriver_ConflictStraddlesSection_Preserved` — boundary edge case
- [x] `TestE2E_MergeDriver_SectionMarkersInsideConflict_Preserved` — markers inside conflict

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/mdsmith/mergedriver.go [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+// fixAtRealPath writes cleaned content to pathname, runs mdsmith
+// fix, copies the result to ours, and restores pathname.
+func fixAtRealPath(cleaned []byte, ours, pathname string) ([]byte, int) {
+	backup, backupErr := os.ReadFile(pathname)
+	if backupErr != nil && !os.IsNotExist(backupErr) {
+		fmt.Fprintf(os.Stderr, "mdsmith: reading %s for backup: %v\n", pathname, backupErr)
+		return nil, 2
+	}
+	if err := os.WriteFile(pathname, cleaned, 0644); err != nil {
+		fmt.Fprintf(os.Stderr, "mdsmith: writing to %s: %v\n", pathname, err)
```
[the remark this answers is not in the document]
Comment (jeduden) [reply cmd/mdsmith/mergedriver.go]: Fixed in 96e006e. `fixAtRealPath` now calls `os.Stat` to capture the original file mode before any writes, and uses it for both the temporary write and restore.

File: cmd/mdsmith/mergedriver.go [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+		}
+		if isRegenSectionEnd(trimmed) {
+			inSection = false
+		}
+
+		if inSection && isConflictMarker(trimmed) {
+			continue
+		}
+
+		out = append(out, line)
```
[the remark this answers is not in the document]
Comment (jeduden) [reply cmd/mdsmith/mergedriver.go]: Fixed in 96e006e. The end-marker check now runs after the line is appended, so the entire range from start to end (inclusive) is treated as inside the section. Further improved in c01596c with full conflict state tracking.

File: cmd/mdsmith/mergedriver.go [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+	}
+	if restoreErr != nil {
+		fmt.Fprintf(os.Stderr, "mdsmith: restoring %s: %v\n", pathname, restoreErr)
+		return fixed, 2
+	}
+
+	if fixErr != nil {
+		fmt.Fprintf(os.Stderr, "mdsmith: fix failed: %v\n", fixErr)
+		return fixed, 2
+	}
```
[the remark this answers is not in the document]
Comment (jeduden) [reply cmd/mdsmith/mergedriver.go]: Fixed in 96e006e. `fixErr` is now checked before writing to `ours`, so broken content from a failed fix is never used as the merge result. The working-tree restore still happens first to keep the tree clean.

File: cmd/mdsmith/main.go [hunk 4]
```diff
@@ -59,12 +59,13 @@ func main() {
...
-  fix       Auto-fix lint issues in place
-  help      Show help for rules and topics
-  metrics   Show and rank shared Markdown metrics
-  init      Generate a default .mdsmith.yml config file
-  version   Print version and exit
+  check          Lint Markdown files (default when given file arguments)
+  fix            Auto-fix lint issues in place
+  help           Show help for rules and topics
+  metrics        Show and rank shared Markdown metrics
+  merge-driver   Git merge driver for catalog files
```
[the remark this answers is not in the document]
Comment (jeduden) [reply cmd/mdsmith/main.go]: Fixed in 96e006e. Updated usage text in `main.go` from "Git merge driver for catalog files" to "Git merge driver for regenerable sections".

File: cmd/mdsmith/mergedriver.go [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+	var out [][]byte
+	inSection := false
+	inConflict := false
+
+	for _, line := range lines {
+		trimmed := bytes.TrimSpace(line)
+
+		if isRegenSectionStart(trimmed) {
+			inSection = true
+		}
```
[the remark this answers is not in the document]
Comment (jeduden) [reply cmd/mdsmith/mergedriver.go]: Addressed by the fix in f4d2780. The `>>>>>>>` close marker is now only stripped when `inConflict` is true. Since `inConflict` is only set to true when `<<<<<<<` is seen while `inSection` is true, a conflict that opens *outside* a section never sets `inConflict`. This means section markers inside such a conflict are left alone, and the conflict markers surrounding them are preserved. Added `TestE2E_MergeDriver_SectionMarkersInsideConflict_Preserved` to cover this.

File: cmd/mdsmith/e2e_test.go [hunk 6]
```diff
@@ -1228,3 +1228,342 @@ func TestE2E_Check_NoArgs_NoConfig_ExitsZero(t *testing.T) {
...
+
+	result, _ := os.ReadFile(ours)
+	content := string(result)
+	if !strings.Contains(content, "=======") {
+		t.Error("setext heading underline (=======) was incorrectly stripped")
+	}
+	if !strings.Contains(content, "Title") {
+		t.Error("expected regenerated include content with Title")
+	}
+}
```
[the remark this answers is not in the document]
Comment (jeduden) [reply cmd/mdsmith/e2e_test.go]: Added in f4d2780. Two tests cover these edge cases: `TestE2E_MergeDriver_ConflictStraddlesSection_Preserved` (conflict opens before section, closes inside) and `TestE2E_MergeDriver_SectionMarkersInsideConflict_Preserved` (section markers inside a conflict block). Both verify all conflict markers are preserved.
